testcafe/tests/pages/photos-model.js

```diff
@@ -61,20 +68,21 @@ export default class Page {
 
   async waitForLoading() {
     await t.expect(this.loading.exists).notOk('Page still loading')
+    await isExistingAndVisibile(this.contentWrapper, 'Content Wrapper')
   }
 
   //@param {string} when : text for console.log
   async getPhotosCount(when) {
     let allPhotosCount
-
-    if (await this.photoEmpty.exists) {
+    if ((await this.folderEmpty.exists) && (await this.folderEmpty.visible)) {
+      await isExistingAndVisibile(this.folderEmpty, 'Photo Empty section')
       allPhotosCount = 0
-    } else if (await this.photoSection.exists) {
-      await t
-        .expect(this.allPhotosWrapper.exists)
-        .ok('No Picture wrapper')
-        .expect(this.allPhotos.exists)
-        .ok('No Picture')
+    } else if (
+      (await this.photoSection.exists) &&
+      (await this.photoSection.visible)
```

Comment:
idem ?

Reply:
Not here ! the 1st one was a double check indeed, but here, I check that there is a `photoSection`. then I check for `allPhotosWrapper` which is another selector.